src/hatch/template/default.py

```diff
@@ -7,14 +10,14 @@
 class DefaultTemplate(TemplateInterface):
     PLUGIN_NAME = 'default'
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
         self.plugin_config.setdefault('ci', False)
         self.plugin_config.setdefault('src-layout', False)
         self.plugin_config.setdefault('tests', True)
 
-    def initialize_config(self, config):
+    def initialize_config(self, config: Dict) -> None:
```

Comment:
Can config be a little more specific? Something like:
```suggestion
    def initialize_config(self, config: Dict[str, Dict]) -> None:
```

Reply:
Sure, originally I had them be very specific but the type hints were quite long then, I can adjust them a bit again.

Reply:
Long typehints are usually not a problem in and of themselves. They are an indicator of bigger issues :D

Reply:
Also, complex dicts can be expressed much better with typeddicts